### PR TITLE
Enable holochain cargo build argument overrides

### LINKED ISSUE #34: Enable Holochain feature overrides

### PR DESCRIPTION
**Pull request by jost-s**
* Extended flake to enable overrides for holochain build
* Added template for a holochain build with custom cargo build arguments

closes #34

### CODE REVIEW

**Comment by jost-s** [comment]
Build of `hcterm` fails because [crane doesn't detect it as a binary as part of `buildPackage`](https://github.com/ipetkov/crane/issues/268). 2 solutions:
1. Replace `buildPackage` with `buildDepsOnly` and `buildPackage` for each package.
2. Add `[[bin]]` section to `hcterm` package.

I want to do 2. Agreed, @ThetaSinner?

File: templates/custom/flake.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    nixpkgs.follows = "holonix/nixpkgs";
+    flake-parts.follows = "holonix/flake-parts";
+  };
+
+  outputs = inputs@{ flake-parts, ... }: flake-parts.lib.mkFlake { inherit inputs; } {
+    systems = builtins.attrNames inputs.holonix.devShells;
+    perSystem = { inputs', pkgs, ... }:
+      let
+        # Disable default features and enable CHC (chain head coordination) feature for holochain package.
+        cargoExtraArgs = "--no-default-features --features chc";
```
Comment (ThetaSinner) [inline templates/custom/flake.nix]: Necessary to disable default here? We should be able to use add the `chc` feature right?

File: templates/custom/flake.nix
Comment (ThetaSinner) [inline templates/custom/flake.nix:1]: New templates need to be listed in the root flake to be usable with `nix flake init -t xxx`. They're also not automatically picked up in CI, so they should be added to the matrix there to be checked

**Comment by ThetaSinner** [comment]
> I want to do 2. Agreed, @ThetaSinner?

Yeah that's fine with me, just one more gap in crane's handling of workspaces I guess!

File: templates/custom/flake.nix (hunk 1 above)
Comment (jost-s) [reply templates/custom/flake.nix]: Not at all, that was from my `sqlite` feature test.

File: templates/custom/flake.nix
Comment (jost-s) [reply templates/custom/flake.nix:1]: Yes, I've added that but not pushed the change.

**Comment by jost-s** [comment]
Okay, now it works. I've tested it with `holochain/holochain/develop`. Once the new holochain release is out, I'll rebase this PR and tests should pass.

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**